src/langchain_google_alloydb_pg/model.py

```diff
@@ -0,0 +1,177 @@
+        result = await self._query_db(db_flag_query)
+        assert (
+            result[0]["setting"] == "on"
+        ), "google_ml_integration.enable_model_support DB Flag not set"
+
+    async def alist_model(
+        self, model_id: str = "textembedding-gecko@001"
+    ) -> List[Sequence]:
+        """Lists the model details for a specific model_id.
+
+        Raises:
+            :class:`DBAPIError <sqlalchemy.exc.DBAPIError>`: if model has not been created.
+        """
+        result = await self._engine._run_as_async(self._alist_model(model_id=model_id))
+        return result
+
+    async def amodel_info_view(self) -> List[Sequence]:
+        """Lists all the models and its details."""
+        results = await self._engine._run_as_async(self._amodel_info_view())
+        return results
+
+    async def acreate_model(self, model_id: str, model_provider: str, **kwargs) -> None:
+        """Creates a custom text embedding model.
+
+        Raises:
+            :class:`DBAPIError <sqlalchemy.exc.DBAPIError>`: if argument names are not the same as those in the SQL function.
+        """
+        await self._engine._run_as_async(
+            self._acreate_model(model_id, model_provider, **kwargs)
+        )
+
+    async def adrop_model(self, model_id: str) -> None:
+        """Removes a text embedding model."""
+        await self._engine._run_as_async(self._adrop_model(model_id))
+
+    def _validate(self) -> None:
+        """Private function to validate prerequisites.
+
+        Raises:
+            Assertion error if google_ml_integration EXTENSION is not 1.3.
+            Assertion error if google_ml_integration.enable_model_support DB Flag not set.
+        """
+        self._engine._run_as_sync(self.validateGoogleMLExtension())
+        self._engine._run_as_sync(self.validateDBFlag())
+
+    async def _query_db(self, query: str) -> List[Sequence]:
+        """Queries the Postgres database through the engine.
+
+        Raises:
+            Exception if the query is not a returning type."""
+        async with self._engine._pool.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            results = result_map.fetchall()
+        return results
+
+    async def _alist_model(
+        self,
+        model_id: str = "textembedding-gecko@001",
+    ) -> List[Sequence]:
+        """Lists the model details for a specific model_id.
+
+        Raises:
+            :class:`DBAPIError <sqlalchemy.exc.DBAPIError>`: if model has not been created.
+        """
+        query = f"""SELECT * from google_ml.list_model('{model_id}')as t(model_id VARCHAR, model_request_url VARCHAR,
+        model_provider google_ml.model_provider, model_type google_ml.model_type, model_qualified_name VARCHAR ,
+        model_auth_type google_ml.auth_type, model_auth_id VARCHAR, input_transform_fn VARCHAR,output_transform_fn VARCHAR)"""
```

Comment:
Reformat for better readability:
```suggestion
        query = f"""SELECT * FROM
  google_ml.list_model('{model_id}')
AS t(model_id VARCHAR,
    model_request_url VARCHAR,
    model_provider google_ml.model_provider,
    model_type google_ml.model_type,
    model_qualified_name VARCHAR,
    model_auth_type google_ml.auth_type,
    model_auth_id VARCHAR,
    input_transform_fn VARCHAR,
    output_transform_fn VARCHAR)"""
```